### ISSUE TITLE
Consider supporting GraalJS' foreign-object-prototype

### ISSUE THREAD

**Issue by pintomau** [bsorrentino/java2typescript#34]
As explained in the documentation https://www.graalvm.org/reference-manual/js/Options/#to-the-launcher

> --js.foreign-object-prototype: provide JavaScript’s default prototype to foreign objects that mimic JavaScript’s own types (foreign Arrays, Objects and Functions). Boolean value, default is false.

So, for example, List<E> could extend Array<E> and so on.

**Comment by bsorrentino** [comment]
Hi @pintomau thanks for feedbak

Really this project is most related to generate Typescript compatible declarations for Java Object useful to make easier interoperability with Java if you'd like use Typescript to develop on top of GraalJS 

Your suggestion is mostly for the GraalJS  runtime and I'll take it in consideration in the examples

**Comment by pintomau** [comment]
Just want to add an example to this.

This flag makes it more natural to work with Java objects from the PoV of someone coming from the JS/TS world.

Instead of working with `stream().map()` or `.get(0)`, you'd simply use `.map()` and `[0]`.

Currently solved this in our code like this:

```typescript
/**
 * This type represents the union of Java's List and Javascript's Array types.
 *
 * It's useful because we're using the js.foreign-object-prototype GraalJS option which
 * wraps the List type with a compatible Array API.
 *
 * @template E the concrete type hold by the list
 */
type JavaListProxy<E> = List<E> & [E]

// @ts-expect-error Java Lists are wrapped with a JS Array compatible type, therefore we're using an intersection type
const technicalEquipments: JavaListProxy<TechEquipmentItem> = vehicleItem.getTechnicalEquipments() ?? []
```

**Comment by pintomau** [comment]
Actually found a better way, it seems.

Create an `overrides.d.ts` file, add your own `interface List<E> extends Array<E>`, and add it last in the `.tsconfig` includes list.

**Comment by bsorrentino** [comment]
Hi @pintomau  could you provide me a more complete example of that so I could integrate it inside processor ?
